\chapter*{Abstract}
\markboth{Abstract}{Abstract}

\Blindtext

% Daily changes in mammalian physiology, known as circadian rhythms, are coordinated by intricate genetic regulatory networks that process information from the environment to prepare the body for upcoming demands. This natural feed-forward controller is essential to maintaining metabolic homeostasis, and compromised circadian rhythms have been linked to the onset of metabolic disease. Since circadian rhythms can be damped by factors such as jet lag, shift work, and high fat diets, numerous studies have focused on elucidating mechanisms which govern circadian amplitude. My work uses techniques from dynamic systems to model circadian oscillations at different scales. By capturing changes in gene expression using ordinary differential equation models, I am able to investigate how pharmacological changes affect gene expression at the single-cell level. However, physiologically relevant changes gene expression are determined by the collective output of many cells at the tissue-level.  I therefore develop computationally efficient techniques of simulating the average response from a population of oscillating cells. These techniques use a probability density function to describe the synchrony of the population, and a convection-diffusion equation to capture the evolution of the population in time. My work provides new tools to help understand the effects of pharmacological perturbations in experimental conditions, which will ultimately allow the development of therapies to treat common health issues associated with the demands of a 24-hr society.
